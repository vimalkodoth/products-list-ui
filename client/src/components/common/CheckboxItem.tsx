/** @jsxImportSource @emotion/react */
import useCheckboxItem from '../../hooks/useCheckboxItem';
import { itemCSS } from './CheckboxItem.styles';

const CheckboxItem = ({
    onChange,
    label,
    isUnChecked = false,
}): JSX.Element => {
    const { handleOnChange, checked } = useCheckboxItem(onChange, isUnChecked);
    return (
        <div css={itemCSS} className={`${checked ? 'checked' : ''}`}>
            <input
                type="checkbox"
                id={label}
                checked={checked}
                onChange={handleOnChange}
            />
            <label htmlFor={label}>{label}</label>
        </div>
    );
};

export default CheckboxItem;
