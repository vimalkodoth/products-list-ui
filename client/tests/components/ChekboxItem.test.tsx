import React from 'react';
import { mount } from 'enzyme';
import CheckboxItem from '../../src/components/common/CheckboxItem';

describe('CheckboxItem', () => {
    it('should render checkbox component', async () => {
        const onChange = jest.fn();
        const wrapper = mount(
            <CheckboxItem onChange={onChange} label={'ProductText'} />
        );
        expect(wrapper.getElements()).toMatchSnapshot();
        expect(wrapper.find('[type="checkbox"]').props().checked).toBeFalsy();
    });
    it('should render checkbox as selected', async () => {
        const onChange = jest.fn();
        const wrapper = mount(
            <CheckboxItem
                onChange={onChange}
                label={'ProductText'}
                isChild={false}
            />
        );
        const checkbox = wrapper.find('[type="checkbox"]');
        expect(wrapper.getElements()).toMatchSnapshot();
        expect(checkbox.props().checked).toBeFalsy();
        checkbox.simulate('change', { target: { checked: true } });
        wrapper.setProps({ isChild: true });
        expect(checkbox.props().checked).toBeFalsy();
    });
});
