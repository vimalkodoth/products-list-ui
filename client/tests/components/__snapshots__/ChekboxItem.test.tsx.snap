// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckboxItem should render checkbox as selected 1`] = `
Array [
  <CheckboxItem
    isUnChecked={false}
    label="ProductText"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            false,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  />,
]
`;

exports[`CheckboxItem should render checkbox component 1`] = `
Array [
  <CheckboxItem
    label="ProductText"
    onChange={
      [MockFunction] {
        "calls": Array [
          Array [
            false,
          ],
        ],
        "results": Array [
          Object {
            "type": "return",
            "value": undefined,
          },
        ],
      }
    }
  />,
]
`;
